### ISSUE TITLE
Card tooltips in Timeline view

### ISSUE THREAD

**Issue by taylan** [stevschmid/trackobot.com#23]
I was thinking it would be nice if card tooltips were shown when the user hovers over a card in the timeline view. I have something like this in mind:

![](http://i.imgur.com/ORa3pQR.png)

I know of two sites that provide Javascript tooltip service for HS cards: [HearthPwn](http://www.hearthpwn.com/tooltips) and [Hearthhead](http://www.hearthhead.com/tooltips). Unfortunately, both of them require you to know the site-specific IDs for cards to construct the url.

**However**, I might have found a bit of a loophole. The url format for Hearthhead card images contains the HearthStone card IDs. Something like:

`http://wow.zamimg.com/images/hearthstone/cards/enus/original/BRMC_91.png`

So we can obtain card images using the `Cards.ref` field. The show/hide and positioning logic can be implemented with some simple Javascript (or we can use a lightweight plugin.)

What do you think? If this sounds like a good idea to you, I would be interested in working on it.

**Comment by stevschmid** [comment]
With the advent of TGT I really like this idea.

It's a bummer that neither Hearthpwn nor Hearthhead allows for tooltips with the card id (`ref`). Just grabbing the image would probably violate the purpose why such tooltip services were created in the first place. I'm not a fan of a mapping either. But I don't have a good solution right now.

**Comment by taylan** [comment]
Yeah, I don't like using the image directly either. I searched for HearthstoneCardID->HearthPwnCardID or similar mappings but I couldn't find anything. We can attempt to create that mapping ourselves.

In the card detail pages in Hearthhead, the HearthstoneCardID is used in several places. (For example go to the Hearthhead page for [Alexstrasza](http://www.hearthhead.com/card=581/alexstrasza), view source and search for `EX1_561`). (Hearthpwn card details pages do not have the HearthstoneCardID values anywhere so Hearthhead is the right choice for this method I suppose.)

We can write a simple spider to crawl all card detail pages in Hearthhead (Start [here](http://www.hearthhead.com/cards) and go through all the pages), and extract the HearthstoneCardIDs from the page HTML. This way, we can have our own Card.ref->HearthheadCardID mapping. Once we have that, we can use the Hearthhead Tooltip thingy legitimately. It would be a lot better than hotlinking to a card image directly.

What do you think? Are any of these worth bothering with? Maybe the Hearthhead guys would give us the mapping if we asked them nicely :) Or maybe one of them would improve their tooltip script and  allow tooltips using the Hearthstone Card ID.

**Comment by emilesilvis** [comment]
Could http://hearthstoneapi.com be a viable option?

**Comment by taylan** [comment]
@emilesilvis it could but it looks like this api also returns image urls like

`http://wow.zamimg.com/images/hearthstone/cards/enus/original/EX1_572.png`

And this API doesn't look like it is affiliated in any way with Hearthhead so they are basically doing the same hacky thing we considered. If Hearthhead is OK with hotlinking in this way then we can do it.

**Comment by stevschmid** [comment]
Feel free to contact the Hearthhead guys. I would love to add tooltips, but the solution must be a robust one.

**Comment by taylan** [comment]
OK I contacted Hearthhead and learned that I was way off with the Hearthstone Card IDs. Here's the card definition for Alexstrasza in the `DBF/CARD.XML` file:

```
    <Record>
		<Field column="ID">581</Field>
		<Field column="NOTE_MINI_GUID">EX1_561</Field>
		<Field column="IS_COLLECTIBLE">True</Field>
		<Field column="HERO_POWER_ID">0</Field>
		<Field column="LONG_GUID">00000011-7ec9-40d6-accf-6aec89ee297f</Field>
		<Field column="CRAFTING_EVENT">always</Field>
	</Record>
```

The `NOTE_MINI_GUID` is the one used in trackobot. Hearthhead uses the `ID` field. But we don't have that ID in cards.json in the app. @stevschmid what's the source of the `db/cards.json` file? If we add the ID field to the Card model we'll have what we need for the Hearthhead tooltips.

**Comment by stevschmid** [comment]
Very cool! Thanks. I will try to add the ID to `Card` in the next update.

**Comment by emilesilvis** [comment]
Nice!
